exe/fluent-diagtool

```diff
@@ -16,11 +16,17 @@
 #    limitations under the License.
 #
 
+on_windows = /mingw/.match?(RUBY_PLATFORM)
+
 require 'optparse'
 require 'fluent/diagtool/collectutils'
 require 'fluent/diagtool/maskutils'
 require 'fluent/diagtool/validutils'
-require 'fluent/diagtool/diagutils'
+if on_windows
+  require 'fluent/diagtool/diagutils-win'
+else
```

Comment:
How about put platform specific code under `fluent/diagtool/diagutils/windows' namespace?

Reply:
Thanks. I haven't figured out how to do it well.
The implementation of this PR is based on the assumption that we don't require `fluent/diagtool/diagutils` on Windows.

It is because we don't use most of the features in `fluent/diagtool/diagutils`.
For such a minimum Windows support, I think we can keep the maintainability of existing codes by separating `diagutils` and `diagutils-win` completely, as the following comment.

https://github.com/daipom/diagtool/blob/0af4ec60843fb1b08d70055c91d0dcc77cef0e2f/lib/fluent/diagtool/diagutils-win.rb#L22-L24

> Splitting the implementation just to easily provide minimal support for Windows.
> If we are going to support more features for Windows, it would be better to remove
> this file and improve the origin file "diagutils.rb".